service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/HearingBookingDetailsController.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import com.google.common.collect.ImmutableList;
+import io.swagger.annotations.Api;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.HearingBooking;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.service.MapperService;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Objects;
+
+@Api
+@RestController
+@RequestMapping("/callback/add-hearing-bookings")
+public class HearingBookingDetailsController {
+    private final MapperService mapperService;
+
+    @Autowired
+    public HearingBookingDetailsController(MapperService mapperService) {
+        this.mapperService = mapperService;
+    }
+
+    @PostMapping("/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody CallbackRequest callbackrequest) {
+        CaseDetails caseDetails = callbackrequest.getCaseDetails();
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetails.getData())
+            .errors(validate(caseDetails))
+            .build();
+    }
+
+    private List<String> validate(CaseDetails caseDetails) {
+        ImmutableList.Builder<String> errors = ImmutableList.builder();
+
+        CaseData caseData = mapperService.mapObject(caseDetails.getData(), CaseData.class);
+
+        caseData.getHearingDetails().stream()
+            .map(Element::getValue)
+            .map(HearingBooking::getHearingDate)
+            .filter(Objects::nonNull)
+            .filter(hearingDate -> hearingDate.isBefore(LocalDate.now()) || hearingDate.equals(LocalDate.now()))
```

Comment:
I think that this would make a nice utility validator function. What do you think?

Reply:
We discussed this yesterday?I think for this story it should remain as is and as we need it then build on it but for this story I think it is adequate?

Reply:
I mean just the filter, maybe no action for now but in the future it may make sense to move this into a filter utility method

Reply:
You could change that isBefore || isToday into !isAfter :)